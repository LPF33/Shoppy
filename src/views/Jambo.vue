<template>
  <section>
    <h1>Jambo</h1>
    <h2 v-if="!counter">Start</h2>
    <h2 v-else>Spielzug: {{ counter }}</h2>
    <div v-if="visual">
      <div v-for="(item, index) in visual" :key="index" class="runde">
        <font-awesome-icon
          icon="fa-duotone fa-card-heart"
          style="
            --fa-secondary-opacity: 1;
            --fa-primary-color: rgb(255, 92, 92);
            --fa-secondary-color: white;
          "
        />
      </div>
    </div>
    <button @click="increment">
      <font-awesome-icon
        icon="fa-duotone fa-cards-blank"
        style="
          --fa-secondary-opacity: 1;
          --fa-primary-color: rgb(246, 90, 131);
          --fa-secondary-color: rgb(127, 183, 126);
        "
      />
    </button>
    <button @click="reset">
      <font-awesome-icon
        icon="fa-duotone fa-eraser"
        style="
          --fa-secondary-opacity: 1;
          --fa-primary-color: red;
          --fa-secondary-color: black;
        "
      />
    </button>
  </section>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "Jambo",
  data() {
    return {
      counter: 0,
    };
  },
  methods: {
    increment() {
      if (this.counter >= 5) {
        this.reset();
        return;
      }
      this.counter++;
    },
    reset() {
      this.counter = 0;
    },
  },
  computed: {
    visual(): any[] | null {
      return this.counter > 0 ? new Array(this.counter) : null;
    },
  },
});
</script>

<style scoped>
section {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  background-color: rgb(255, 232, 152);
  user-select: none;
}

h1,
h2 {
  margin: 10px 0;
}

button {
  display: block;
  font-size: 3rem;
  height: 6rem;
  width: 6rem;
  border: 1px solid grey;
  border-radius: 50%;
  background-color: transparent;
  line-height: 1.5;
  padding: 10px;
  cursor: pointer;
  margin-top: 2rem;
  user-select: none;
  box-shadow: 0 0 8px black;
}

.runde {
  font-size: 3rem;
}
</style>
