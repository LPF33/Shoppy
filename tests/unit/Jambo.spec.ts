import { mount } from "@vue/test-utils";
import Jambo from "@/views/Jambo";

describe("Testing the Jambo component", () => {
  it("Initial Jambo component is mounted correctly", () => {
    const wrapper = mount(Jambo);
    expect(wrapper.html()).toMatchSnapshot();

    const jamboSection = wrapper.find(".jambo-section");
    expect(jamboSection.exists()).toBeTruthy();
    expect(jamboSection.find("h1").text()).toBe("Jambo");
    expect(jamboSection.find("h2").text()).toBe("Start");
    expect(jamboSection.findAll("button")).toHaveLength(2);
    expect(jamboSection.findAll("button")[0].text()).toBe("âž•");
    expect(jamboSection.findAll("button")[1].text()).toBe("âšª");
  });

  it("Click on first button, should increment counter", async () => {
    const wrapper = mount(Jambo);

    const incrementButton = wrapper.findAll(".jambo-section > button")[0];
    await incrementButton.trigger("click");
    expect(wrapper.find("h2").text()).toBe("Spielzug: 1");
    expect(wrapper.findAll("section")).toHaveLength(1);
    expect(wrapper.findAll("section > .runde")).toHaveLength(1);
    expect(wrapper.find("section > .runde").text()).toBe("ðŸŽ²");
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Click on first button 3 times , should increment counter", async () => {
    const wrapper = mount(Jambo);

    const incrementButton = wrapper.findAll(".jambo-section > button")[0];
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    expect(wrapper.find("h2").text()).toBe("Spielzug: 3");
    expect(wrapper.findAll("section > .runde")).toHaveLength(3);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Click on first button 6 times, should reset counter", async () => {
    const wrapper = mount(Jambo);

    const incrementButton = wrapper.findAll(".jambo-section > button")[0];
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");

    expect(wrapper.find("h2").text()).toBe("Spielzug: 5");
    expect(wrapper.findAll("section > .runde")).toHaveLength(5);
    await incrementButton.trigger("click");

    expect(wrapper.find("h2").text()).toBe("Start");
    expect(wrapper.find("section").exists()).toBeFalsy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Click on first button 4 times, click on reset button, should reset counter", async () => {
    const wrapper = mount(Jambo);

    const incrementButton = wrapper.findAll(".jambo-section > button")[0];
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");
    await incrementButton.trigger("click");

    expect(wrapper.find("h2").text()).toBe("Spielzug: 4");
    expect(wrapper.findAll("section > .runde")).toHaveLength(4);

    const resetButton = wrapper.findAll(".jambo-section > button")[1];
    await resetButton.trigger("click");
    expect(wrapper.find("h2").text()).toBe("Start");
    expect(wrapper.find("section").exists()).toBeFalsy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
