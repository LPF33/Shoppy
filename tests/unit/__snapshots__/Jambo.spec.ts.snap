// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Jambo component Click on first button 3 times , should increment counter 1`] = `
<section>
  <h1>Jambo</h1>
  <h2>Spielzug: 3</h2>
  <div>
    <div class="runde">
      <font-awesome-icon icon="fa-duotone fa-card-heart" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(255, 92, 92); --fa-secondary-color: white;"></font-awesome-icon>
    </div>
    <div class="runde">
      <font-awesome-icon icon="fa-duotone fa-card-heart" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(255, 92, 92); --fa-secondary-color: white;"></font-awesome-icon>
    </div>
    <div class="runde">
      <font-awesome-icon icon="fa-duotone fa-card-heart" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(255, 92, 92); --fa-secondary-color: white;"></font-awesome-icon>
    </div>
  </div><button>
    <font-awesome-icon icon="fa-duotone fa-cards-blank" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(246, 90, 131); --fa-secondary-color: rgb(127, 183, 126);"></font-awesome-icon>
  </button><button>
    <font-awesome-icon icon="fa-duotone fa-eraser" style="--fa-secondary-opacity: 1; --fa-primary-color: red; --fa-secondary-color: black;"></font-awesome-icon>
  </button>
</section>
`;

exports[`Testing the Jambo component Click on first button 4 times, click on reset button, should reset counter 1`] = `
<section>
  <h1>Jambo</h1>
  <h2>Start</h2>
  <!--v-if--><button>
    <font-awesome-icon icon="fa-duotone fa-cards-blank" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(246, 90, 131); --fa-secondary-color: rgb(127, 183, 126);"></font-awesome-icon>
  </button><button>
    <font-awesome-icon icon="fa-duotone fa-eraser" style="--fa-secondary-opacity: 1; --fa-primary-color: red; --fa-secondary-color: black;"></font-awesome-icon>
  </button>
</section>
`;

exports[`Testing the Jambo component Click on first button 6 times, should reset counter 1`] = `
<section>
  <h1>Jambo</h1>
  <h2>Start</h2>
  <!--v-if--><button>
    <font-awesome-icon icon="fa-duotone fa-cards-blank" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(246, 90, 131); --fa-secondary-color: rgb(127, 183, 126);"></font-awesome-icon>
  </button><button>
    <font-awesome-icon icon="fa-duotone fa-eraser" style="--fa-secondary-opacity: 1; --fa-primary-color: red; --fa-secondary-color: black;"></font-awesome-icon>
  </button>
</section>
`;

exports[`Testing the Jambo component Click on first button, should increment counter 1`] = `
<section>
  <h1>Jambo</h1>
  <h2>Spielzug: 1</h2>
  <div>
    <div class="runde">
      <font-awesome-icon icon="fa-duotone fa-card-heart" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(255, 92, 92); --fa-secondary-color: white;"></font-awesome-icon>
    </div>
  </div><button>
    <font-awesome-icon icon="fa-duotone fa-cards-blank" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(246, 90, 131); --fa-secondary-color: rgb(127, 183, 126);"></font-awesome-icon>
  </button><button>
    <font-awesome-icon icon="fa-duotone fa-eraser" style="--fa-secondary-opacity: 1; --fa-primary-color: red; --fa-secondary-color: black;"></font-awesome-icon>
  </button>
</section>
`;

exports[`Testing the Jambo component Initial Jambo component is mounted correctly 1`] = `
<section>
  <h1>Jambo</h1>
  <h2>Start</h2>
  <!--v-if--><button>
    <font-awesome-icon icon="fa-duotone fa-cards-blank" style="--fa-secondary-opacity: 1; --fa-primary-color: rgb(246, 90, 131); --fa-secondary-color: rgb(127, 183, 126);"></font-awesome-icon>
  </button><button>
    <font-awesome-icon icon="fa-duotone fa-eraser" style="--fa-secondary-opacity: 1; --fa-primary-color: red; --fa-secondary-color: black;"></font-awesome-icon>
  </button>
</section>
`;
